google-startup-scripts/usr/share/google/set-hostname

```diff
@@ -15,7 +14,7 @@
 
 # Deal with a new hostname assignment.
 
-if [[ -n "$new_host_name" && -n "$new_ip_address" ]]; then
+if [ -n "$new_host_name" -a -n "$new_ip_address" ]; then
```

Comment:
If we're trying for maximum portability, could you do ```[ -n "$new_host_name" ] && [ -n "$new_ip_address" ]``` instead of -a (and test again)? The -a flag is an XSI extension according to SUSV3: http://pubs.opengroup.org/onlinepubs/009695399/utilities/test.html :)